src/intelligence_layer/connectors/document_index/document_index.py

```diff
@@ -29,11 +29,20 @@ class IndexConfiguration(BaseModel):
 
     Args:
         embedding_type: "symmetric" or "asymmetric" embedding type.
+        chunk_overlap: The maximum number of tokens of overlap between consecutive chunks. Must be
+            less than `chunk_size`.
         chunk_size: The maximum size of the chunks in tokens to be used for the index.
     """
 
     embedding_type: Literal["symmetric", "asymmetric"]
-    chunk_size: int
+    chunk_overlap: int = Field(default=0, ge=0)
+    chunk_size: int = Field(..., gt=0, le=2046)
```

Comment:
are you sure its 2046 and not 2047?

Reply:
Yes - see the schema at https://app.document-index.aleph-alpha.com/api